profiler/src/util/macos.rs

```diff
@@ -0,0 +1,6 @@
+use std::{error::Error, fs::File, path::Path};
+
+pub(crate) fn open_without_cache(path: &Path) -> Result<File, Box<dyn Error>> {
+    let file = File::open(path)?;
+    Ok(file)
```

Comment:
I assume this should work:

```rust
    let ret = unsafe {
        libc::fcntl(
            file.as_raw_fd(),
            libc::F_NOCACHE,
            1 as libc::c_int,
        )
    };
```

based on: https://stackoverflow.com/a/2307055

Reply:
Good hint! 

However, according to https://developer.apple.com/forums/thread/25464, `F_NOCACHE` will avoid caching the file, but it doesn't remove it from the cache. I need to think about preventing caching or dropping the file from the cache.